### PR TITLE
Improved Chameleon Disguise Kit for Syndiborgs

### PR DESCRIPTION
**Pull request by Surrealaser**
:cl:
tweak: Syndiborg chameleon disguise kit now allows its user to disguise as any standard Nanotrasen borg type.
add: Medical Syndiborg loses its emag (would be too strong with the disguise kit) and roller bed, and gains the chameleon disguise kit, dropper, syringe, beaker apparatus, hugging, holobed projector and treat dispenser.
balance: Morphine removed from Syndiborg Hypospray; replaced with heparine for lethals, and lipolicide and anacea for stealthier and delayed kills that require more effort and time investment to deal with.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
um

I don't think we should give it to syndicate mediborgs for the same reason that we don't let ops buy sleepypens and chameleon kits anymore. It'd be way too easy for a syndicate mediborg to disguise themselves as a mediborg, walk up to the captain, and bop him with a morphine injection or two from their hypospray for an EZ clap.

Being unable to subvert other borgs is also a big blow to syndicate mediborgs, and I honestly don't think that losing the emag is worth gaining any of the things you gave syndicate mediborgs here.

In short, leave the disguising, infiltration, and subtle sabotage to the saboteur borgs (it's pretty much their entire schtick) and let the syndicate mediborgs stick to their role of healing (and opening doors for) the human operatives that they're supposed to be sticking with (instead of solo blitzing the captain).

**Comment by ATH1909** [comment]
Also, what exactly do you see syndicate medical cyborgs using this chameleon disguise kit for (that isn't a cancerous blitz strategy)? I'm genuinely curious.

**Comment by Surrealaser** [comment]
@ATH1909 Assault borgs still have emags, so if you want to go around emagging station borgs, you have that option.

Also is chameleon kit + morphine so good that it's a serious balance concern or is it not nearly as strong as the status quo of emag without disguises? Which is it?

Lastly, I've been considering alternate chem loadouts for the syndiborg hypo.

**Comment by ATH1909** [comment]
> Also is chameleon kit + morphine so good that it's a serious balance concern or is it not nearly as strong as the status quo of emag without disguises? Which is it?

The chameleon disguise kit is really, really strong (for syndicate medical cyborgs) if you're going with a specific cheap strategy (disguising, injecting the captain, running away (or bolting him in somewhere), then using your nuke disk pinpointer to find his unconscious body), but way, way worse than an emag if you're not. If you're not going alone, what's the point of disguising when the people you're hanging with are clearly nuke ops? If you ARE going alone, what can you really do other than score a cheap captain kill?

The syndicate mediborg's emag synergizes with the rest of its kit. A nuke op team hires a syndicate medical cyborg because it can perform two important roles at once: it can heal the team and help the team counter the efforts of other silicons. The emag helps greatly with the latter role. You can use it to keep open doors that enemy silicons keep closing, seal bolted doors with it to make them much more of an obstacle (even to enemy silicons and people with tools), and, of course, subvert enemy silicons who lose melee fights with you (and when you're with your team, you can risk a melee fight with an enemy borg much more confidently, since your team can bail you out if you get flashed first).

The disguise kit, on the other hand, clashes completely with the rest of the syndicate medical cyborg's kit. You're either depending upon it/abusing it (splitting off from the rest of your nuke op team to go and get solo stealth kills), not using it at all (sticking with the team), or artificially extending the length of the round with it (by using it as a disguise after the rest of your team has died and you have no hope of winning). It doesn't help the syndicate medical cyborg do either of the two aforementioned roles (healing the team and helping the team counter enemy silicons), instead presenting a new potential role for the mediborg that is mutually exclusive with the "healing the team" role: solo stealth assassin, which I'm not sure is a role we want to be promoting for nuke op teams.

**Comment by ATH1909** [comment]
Also, how in the hell is this a "tweak:"?!

**Comment by Surrealaser** [comment]
@ATH1909 as with the saboteur, the point of exploiting the syndimedborg's disguise kit is to cause chaos and fuck with the station's capabilities, albeit from a different vantage, and with different tools, and you can seamlessly switch between doing stealth assassinations and sabotage (toxins in cryo, melting key objects and machines with acid, expending chem charge (in order to make say toxic foam bombs, because you can now), mercy killing the wounded and undermining medbay's efforts in general), and moving to support the team when the main force arrives. As I said before, I'm reviewing the reagent list of the syndicate hypospray for suitability to see if there are better options. Beyond that, having that flexibility is more than worth the loss of the emag in my view.

I think the real problem here is that a piercing hypo with morphine is pretty cancerous and makes stealth kills altogether too easy.

**Comment by ATH1909** [comment]
> @ATH1909 as with the saboteur, the point of exploiting the syndimedborg's disguise kit is to cause chaos and fuck with the station's capabilities, albeit from a different vantage, and with different tools, and you can seamlessly switch between doing stealth assassinations and sabotage (toxins in cryo, melting key objects and machines with acid, expending chem charge (in order to make say toxic foam bombs, because you can now), mercy killing the wounded and undermining medbay's efforts in general), and moving to support the team when the main force arrives. As I said before, I'm reviewing the reagent list of the syndicate hypospray for suitability to see if there are better options. Beyond that, having that flexibility is more than worth the loss of the emag in my view.

Proposal, then: instead of giving the syndicate medical cyborg those beaker manipulators, droppers, and syringes, give syndicate *saboteur* cyborgs those modules. Since they're no longer going to be restricted to only using an engiborg disguise after this PR and there's precedent for them having non-engiborg tools (see: their cable cuffs), we can give them the mediborg tools they need to effectively sabotage Medbay in the ways you described, which I think would be wonderful and give syndicate saboteur borgs more to do after they kill the AI and sabotage the SM.

Instead of trying to ram the metaphorical circular rods that are syndicate mediborgs into a square hole of Medbay sabotage, we should be trying to give tools to the metaphorical rectangular rods that are syndicate saboteur cyborgs that'd let them fit satisfyingly into the aforementioined square hole of Medbay sabotage. That was a terrible metaphor, but I think the point is clear: these tools that you're giving to syndicate medical cyborgs to allow them to sabotage Medbay could be just as easily given to syndicate saboteur cyborgs instead, and I think they'd fit much better in the toolkits of syndicate saboteur cyborgs than they would in the toolkits of syndicate medical cyborgs.

**Comment by ATH1909** [comment]
tldr;

give this stuff to sabby borgs, they're the ones who are supposed to be sabotaging things

**Comment by Surrealaser** [comment]
@ATH1909 I think you're _way_ too hung up on the title.

Let's face it, the 'saboteur' borg is really a syndi-engiborg, just as the syndi-mediborg is what it says on the tin.

They're fine for sabotaging the space station in two very different ways. Consolidating both types of sabotage into one borg would mean an insanely bloated module list, or would require reworking the module list to use a very specific set of tools.

I think what I will do is rework the medical syndiborg hypospray chems to replace morphine with less cancerous chems.

**Comment by ATH1909** [comment]
> Consolidating both types of sabotage into one borg would mean an insanely bloated module list, or would require reworking the module list to use a very specific set of tools.

If we condense some of the tools of the syndicate saboteur cyborg into their upgraded versions (crowbar and wirecutters->jaws of life, syndicate screwdriver and wrench->(syndicate?) hand drill), adding a syringe, beaker apparatus, and dropper to syndicate saboteur borgs will only add one extra slot's worth of bloat.

Keep in mind that the syndicate mediborg currently has 19 tools, while the syndicate saboteur cyborg currently has 21 tools. This PR would bring syndicate mediborgs up to a whopping 24 tools, while my proposed way of adding these tools to saboteur borgs would bring them up to only 22 tools (or 24 tools if you don't condense some of their existing tools).

> I think you're way too hung up on the title.

I could say the same thing to you, seeing as how you're trying to ram syndicate medical cyborgs into an "infiltrate and sabotage medbay" role (and keeping syndicate saboteur cyborgs out of that role) just because they have "medical" in their name.

> Let's face it, the 'saboteur' borg is really a syndi-engiborg, just as the syndi-mediborg is what it says on the tin.

Then why'd you give it the ability to disguise itself as non-engiborg modules if you're only going to give it the tools to let it pretend to be an engiborg? Yes, the syndicate saboteur cyborg is SHAPED like a engiborg, but that doesn't mean that its tools have to be restricted to just engineering-related sabotage. I'd again like to point to the cuffs that saboteur borgs get that normal engiborgs (and even the other two types of syndiborg) don't get; a lot of their tools are also used by engiborgs, yes, but that's because a lot of the tools that engiborgs get can be used for sabotage-related purposes.

**Comment by Surrealaser** [comment]
@ATH1909 Many of the tool increments for the medborg are to allow it to pull off a more convincing disguise. Yes, multitools could help keep bloat down with a consolidated sabo-borg, but it still needs finishing touches to help the authenticity of its disguise, which will increase bloat well beyond what you suggest.

Second, no, the essence of the medborg is found not only in its name but its loadout as well.

Third, the expansion of the disguise kit for the saboteur is there both because it may as well utilize an existing system versus snowflaking for reduced functionality, and so it can adopt another form in a pinch as a disposable disguise when it's en route or escaping versus engaging in prolonged interaction.

**Comment by ATH1909** [comment]
> @ATH1909 Many of the tool increments for the medborg are to allow it to pull off a more convincing disguise. Yes, multitools could help keep bloat down with a consolidated sabo-borg, but it still needs finishing touches to help the authenticity of its disguise, which will increase bloat well beyond what you suggest.

Their disguises are SUPPOSED to not hold up to scrutiny (partially because they don't NEED to stand up to scrutiny). If a saboteur borg gets hit once by anything with their disguise up, their disguise drops. Syndiborgs are supposed to get by on just silently acting like they belong where they are and know what they're doing, and that usually works to fool most of the crew. Think of syndiborgs as being more like Agent 47 (from the Hitman series) than a changeling. You don't need to have a health analyzer, disabler, peace hypospray, etc. as a saboteur borg because if someone waves you down and orders you to show them your cyborg (non-syndicate) screwdriver (or another distinctive tool that syndicate saboteur borg's disguise would have but the saboteur borg itself does not), your cover's blown anyway (unless you just silently run away and hope that your accuser just thinks that you didn't hear them).

> Second, no, the essence of the medborg is found not only in its name but its loadout as well.

Fair point there, the surgery tools (aside from, say, the energy saw, which has a force of 30 when on) in the syndicate medical cyborg's list of tools are kind of bloat currently for operative-aligned syndicate mediborgs (since ops have both surgical tools on their shuttle and microbomb implants), but I think they're mostly just there for crew-aligned or wizard-created syndicate mediborgs. Perhaps I should make a PR to do something about that...

> Third, the expansion of the disguise kit for the saboteur is there both because it may as well utilize an existing system versus snowflaking for reduced functionality, and so it can adopt another form in a pinch as a disposable disguise when it's en route or escaping versus engaging in prolonged interaction.

Fair enough. I just feel like we could take advantage of this opportunity to improve the ability of syndicate saboteur cyborgs to sabotage the station.

File: code/FulpstationCode/fulp_syndiborg_files/fulp_syndicate_attack_borg.dm [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/obj/item/robot_module/light_syndicate
+	name = "Syndicate Assault Drone"
+	basic_modules = list(
+		/obj/item/assembly/flash/cyborg,
+		/obj/item/gun/energy/laser/cyborg,
```
Comment (ATH1909) [inline code/FulpstationCode/fulp_syndiborg_files/fulp_syndicate_attack_borg.dm]: uh...? what's this stuff about?

**Comment by Surrealaser** [comment]
@ATH1909 

I want them to be able to stand up to scrutiny; I want Medborgs and Saboborgs alike to be able to feign their disguises convincingly. I want players to be able to go into deep cover and strike when the station is most vulnerable or when a prime opportunity presents itself as they do what they can to disrupt in the meanwhile.

File: code/zFulpstationCode/fulp_overwrite_vars.dm [hunk 2]
```diff
@@ -339,4 +339,54 @@
...
+		/obj/item/pinpointer/syndicate_cyborg,
+		/obj/item/stack/medical/gauze/cyborg,
+		/obj/item/gun/medbeam,
+		/obj/item/borg/cyborghug/medical,
+		/obj/item/borg/lollipop,
+		/obj/item/borg_chameleon
+		)
+
+/obj/item/reagent_containers/borghypo/syndicate
+	desc = "An experimental piece of Syndicate technology used to produce powerful restorative nanites used to very quickly restore injuries of all types. Also synthesizes potassium iodide for radiation poisoning and lipolicide and anacea for stealthy kills and heparin for more overt ones."
```
Comment (ATH1909) [inline code/zFulpstationCode/fulp_overwrite_vars.dm:339]: You do realize that potassium iodide DECREASES (not increases) your radiation level, right? It's a MEDICINE, not a poison.

File: code/FulpstationCode/borg_radio_files/borg_radio.dm [hunk 3]
```diff
@@ -67,7 +67,10 @@
 		return RADIO_CHANNEL_SERVICE
 	if(istype(module, /obj/item/robot_module/syndicate))
 		return RADIO_CHANNEL_SYNDICATE
-
+	if(istype(module, /obj/item/robot_module/syndicate_medical))
```
Comment (ATH1909) [inline code/FulpstationCode/borg_radio_files/borg_radio.dm:70]: ...?

Comment (Surrealaser) [reply code/FulpstationCode/borg_radio_files/borg_radio.dm:70]: I noticed it's something I forgot to account for during the radio overhaul.

File: code/zFulpstationCode/fulp_overwrite_vars.dm (hunk 2 above)
Comment (Surrealaser) [reply code/zFulpstationCode/fulp_overwrite_vars.dm:339]: You do realize that's a default medicine for the syndihypo, right?

Comment (ATH1909) [reply code/zFulpstationCode/fulp_overwrite_vars.dm:339]: Oh, I misinterpretted "synthesizes potassium iodide for radiation poisoning" as you saying that it was FOR CAUSING radiation poisoning, not you saying that it's FOR CURING radiation poisoning.

Maybe you could make that a bit more clear in the description?

**Comment by ATH1909** [comment]
> @ATH1909
> 
> I want them to be able to stand up to scrutiny; I want Medborgs and Saboborgs alike to be able to feign their disguises convincing. I want players to be able to go into deep cover and strike when the station is most vulnerable or when a prime opportunity presents itself as they do what they can to disrupt in the meanwhile.

Okay, but then your syndicate mediborg isn't with your team and isn't using 75% of their toolkit. Does that not at all sound like it might not be the best design choice to you?

**Comment by Surrealaser** [comment]
@ATH1909 If you opt for infiltration gameplay, you're not going to be with your team, but once the team has arrived, you have the flexibility to switch and support them.

File: code/modules/antagonists/nukeop/equipment/borgchameleon.dm [hunk 4]
```diff
@@ -79,7 +81,7 @@
 			animate(offset=f:offset-1, time=rand()*20+10)
 		if (do_after(user, 50, target=user) && user.cell.use(activationCost))
 			playsound(src, 'sound/effects/bamf.ogg', 100, TRUE, -6)
-			to_chat(user, "<span class='notice'>You are now disguised as the Nanotrasen engineering borg \"[friendlyName]\".</span>")
+			to_chat(user, "<span class='notice'>You are now disguised as the Nanotrasen [disguise_text] borg \"[friendlyName]\".</span>")  //FULPSTATION SYNDICATE MEDBORG UPDATE by Surrealistik March 2020
```
Comment (ATH1909) [inline code/modules/antagonists/nukeop/equipment/borgchameleon.dm]: you never defined disguise_text anywhere

File: code/FulpstationCode/fulp_syndiborg_files/fulp_syndicate_medborg.dm [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+			disguise_text = "miner"
+			disguise = pick("miner","minerOLD","spidermin")
+		if("Service")
+			disguise_text = "service"
+			disguise = pick("service_f", "service_m", "brobot", "kent", "tophat")
+		if("Standard")
+			disguise_text = "standard"
+			disguise = pick("robot")
+		if("Cancel")
+			to_chat(user, "<span class='warning'>You aborted disguising.</span>")
```
Comment (ATH1909) [inline code/FulpstationCode/fulp_syndiborg_files/fulp_syndicate_medborg.dm]: "arborted" doesn't seem like the right word to use here

honestly, I think you can do without this message entirely

File: code/modules/antagonists/nukeop/equipment/borgchameleon.dm [hunk 6]
```diff
@@ -62,6 +62,8 @@
 			to_chat(user, "<span class='notice'>\the [src] is recharging.</span>")
 			return
 		animation_playing = TRUE
+		if(!fulp_borg_chameleon_menu(user)) //FULPSTATION SYNDICATE MEDBORG UPDATE by Surrealistik March 2020
```
Comment (ATH1909) [inline code/modules/antagonists/nukeop/equipment/borgchameleon.dm:64]: this should come BEFORE the if statement that says that the projector is still recharging, or else you can't ever turn your projector back on again (it'll always be "recharging") if it turns off (under certain conditions? idk, that was just what I found during testing)

File: code/modules/antagonists/nukeop/equipment/borgchameleon.dm [hunk 7]
```diff
@@ -62,6 +62,8 @@
 			to_chat(user, "<span class='notice'>\the [src] is recharging.</span>")
```
Comment (ATH1909) [inline code/modules/antagonists/nukeop/equipment/borgchameleon.dm:62]: you might want to remove the "\t" here and replace it with a "T"

it's not your code here, but still.

Comment (Surrealaser) [reply code/modules/antagonists/nukeop/equipment/borgchameleon.dm:62]: I would rather not mess with default TG code unless strictly necessary.

File: code/modules/antagonists/nukeop/equipment/borgchameleon.dm (hunk 6 above)
Comment (Surrealaser) [reply code/modules/antagonists/nukeop/equipment/borgchameleon.dm:64]: I shifted it above animation_playing

**Comment by Surrealaser** [comment]
@TheSwain Brought up to list standard and good to go.

**Comment by TheSwain** [comment]
I'm moving /obj/item/organ_storage back to the bottom of the list so there's no chance TG makes a change and it ends up appearing in that list twice. Then merge time.

**Pull request merged**